spec/basic.spec.js

```diff
@@ -2799,4 +2799,33 @@ describe('HtmlWebpackPlugin', () => {
       done();
     });
   });
+
+  it('allows __dirname and __filename in requireExtensions', done => {
+    testHtmlPlugin({
+      mode: 'production',
+      target: 'electron-renderer',
+      entry: path.join(__dirname, 'fixtures/index.js'),
+      output: {
+        path: OUTPUT_DIR,
+        filename: 'index_bundle.js'
+      },
+      plugins: [
+        {
+          apply (compiler) {
+            compiler.hooks.compilation.tap(
+              'some-require-hook-plugin',
+              (compilation) => {
+                compilation.mainTemplate.hooks.requireExtensions.tap(
+                  'some-require-hook',
+                  (source, chunk) => {
+                    return `const something = __dirname + __filename;`;
+                  }
+                );
+              }
+            );
+          }
+        },
```

Comment:
Please remove the hook - we can have a simple template like this:

dirname.ejs
```
<html>
<body>
  dirname: <%= __dirname %>
</body>
</html>
```

Reply:
I've just tried this and the test does not fail when I comment out the addition of `__dirname` to `vm.createContext` so it doesn't actually test the code path which causes the issue.